File: src/main/java/burrow/furniture/hoard/command/ExistCommand.java
package burrow.furniture.hoard.command;

import burrow.core.command.Command;
import burrow.core.command.CommandContext;
import burrow.core.command.CommandType;
import burrow.core.common.ColorUtility;
import burrow.core.common.Values;
import burrow.furniture.hoard.HoardFurniture;
import org.jetbrains.annotations.NotNull;
import picocli.CommandLine;

@CommandLine.Command(
    name = "exist",
    description = "Check if an entry exists."
)
@CommandType(HoardFurniture.COMMAND_TYPE)
public class ExistCommand extends Command {
    @CommandLine.Parameters(index = "0", description = "The id of the entry.")
    private Integer id;

    public ExistCommand(@NotNull final CommandContext commandContext) {
        super(commandContext);
    }

    @Override
    public Integer call() {
        final var hoardFurniture = use(HoardFurniture.class);
        final var isExist = hoardFurniture.getHoard().exist(id);
        final var string = Values.Bool.stringify(isExist);

        buffer.append(ColorUtility.render(string, ColorUtility.Type.KEYWORD));

        return CommandLine.ExitCode.OK;
    }
}

File: src/main/kotlin/furnishing/hoard/command/ExistCommand.kt
package burrow.furnishing.hoard.command

import burrow.furnishing.hoard.Hoard
import burrow.kernel.command.Command
import burrow.kernel.command.CommandData
import picocli.CommandLine

@CommandLine.Command(
    name = "exist",
    description = [""]
)
class ExistCommand(data: CommandData) : Command(data) {
    @CommandLine.Parameters(index = "0", description = ["The ID of the entry."])
    private var id: Int = 0

    override fun call(): Int {
        if (id <= 0) {
            stderr.println("Invalid entry ID: $id")
            return CommandLine.ExitCode.USAGE
        }

        val isExist = use(Hoard::class).exists(id)
        stdout.println(if (isExist) "true" else "false")

        return CommandLine.ExitCode.OK
    }
}